===== Patch instructions generated by SAVER =====
 [+] { Insert: if ({ PortObjectAddPortObject:n$11 == { -1 } }) free(*(PortObjectAddPortObject:poi)) at 11 (line 577, column 12) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
@@ -575,6 +575,8 @@ int PortObjectAddPortObject(PortObject * podst, PortObject * posrc, int *errflag
     {
         PortObjectItem *poi = PortObjectItemDup(po);
         if((ret = PortObjectAddItem(podst, poi, errflag)) != 0)
+            if (ret == -1)
+                free(poi);
             return ret;
     }
 
